### PR TITLE
(#1538) - replace the constructor to use an iterable

### LINKED ISSUE #1537: Paged.java:42-45: We want to use {@code Iterator<X>}...
The puzzle `1464-68f0eee6` from #1464 has to be resolved: 

https://github.com/yegor256/cactoos/blob/86ad37b671edb86a5d92d00c1005cc6c62f2c825/src/main/java/org/cactoos/iterator/Paged.java#L42-L45

The puzzle was created by @baudoliver7 on 19-Jan-21. 

Estimate: 30 minutes,  role: DEV. 

If you have any technical questions, don't ask me, submit new tickets instead. The task will be \"done\" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

### PR DESCRIPTION
**Pull request by SergiusAC**
The PR is related to #1538, but also has solution for #1537, because they are interconnected.

### CODE REVIEW

**Comment by victornoel** [comment]
@SergiusAC could you remove the keyword "resolves" from the description? If you leave it, github will auto close the issue when this PR will be merged and we are relying on 0pdd or bug submitter for closing tickets :)

A REV will soon be assigned by 0crat to your PR by the way, sometimes he is a bit slow to react ;)

**Comment by SergiusAC** [comment]
@victornoel Thanks, I edited the description.

**Comment by victornoel** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/yegor256/cactoos/issues/1596#issuecomment-832050437))

@victornoel This is what I know about this job in C63314D6Z, as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#1596](https://github.com/yegor256/cactoos/issues/1596) is [in scope](http://www.zerocracy.com/policy.html#14) for 1days
  * The role is `REV`
  * The job is assigned to @andreoss[/z](https://www.0crat.com/u/andreoss) for [21hr](http://www.zerocracy.com/policy.html#8)
  * There is no monetary reward attached, it's a [free](http://www.zerocracy.com/policy.html#2) job
  * The job doesn't have any [impediments](http://www.zerocracy.com/policy.html#9)
  * The [budget](http://www.zerocracy.com/policy.html#4) is 15 minutes/points
  * These users are banned and won't be assigned:
    * @sergiusac[/z](https://www.0crat.com/u/sergiusac): This user reported the ticket
  * Job [footprint](https://www.0crat.com/footprint/C63314D6Z?q=%7Bjob%3A%27gh%3Ayegor256%2Fcactoos%231596%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

File: src/main/java/org/cactoos/iterable/Paged.java [hunk 1]
```diff
@@ -46,14 +42,17 @@
...
-    public <I extends Iterator<X>> Paged(
-        final I first, final Func<I, I> next
+    public Paged(
+        final Iterable<? extends X> first,
+        final Func<? super Iterable<? extends X>, ? extends Iterable<? extends X>> next
     ) {
         super(
             new IterableOf<>(
-                new org.cactoos.iterator.Paged<>(first, next)
+                new org.cactoos.iterator.Paged<X>(
```
Comment (andreoss) [inline src/main/java/org/cactoos/iterable/Paged.java:52]: @SergiusAC It's better to use `Scalar<Iterator>` here (see #1592)

**Review by andreoss — CHANGES_REQUESTED** [review]

File: src/main/java/org/cactoos/iterable/Paged.java (hunk 1 above)
Comment (SergiusAC) [reply src/main/java/org/cactoos/iterable/Paged.java]: Done it.

**Review by andreoss — APPROVED** [review]

**Comment by SergiusAC** [comment]
@victornoel My PR was approved by @andreoss

**Comment by victornoel** [comment]
@SergiusAC thx

**Comment by victornoel** [comment]
@rultor merge

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Pull request merged**